racket/src/schemify/schemify.rkt

```diff
@@ -838,17 +838,28 @@
                                     prim-knowns knowns imports mutated simples unsafe-mode?)]
                 [else #f]))
             (define (inline-struct-predicate k s-rator im args)
-              (define type-id (and (known-struct-predicate-authentic? k)
-                                   (pair? args)
+              (define type-id (and (pair? args)
                                    (null? (cdr args))
                                    (inline-type-id k im add-import! mutated imports)))
               (cond
-                [type-id
+                [(not type-id) #f]
+                [(known-struct-predicate-authentic? k)
                  (define tmp (maybe-tmp (car args) 'v))
                  (define ques `(unsafe-struct? ,tmp ,(schemify type-id 'fresh)))
                  (wrap-tmp tmp (car args)
                            ques)]
-                [else #f]))
+                [else
+                 (define tmp (maybe-tmp (car args) 'v))
+                 (define schemified-type-id (schemify type-id 'fresh))
+                 (define tmp-type-id (maybe-tmp schemified-type-id 'v))
```

Comment:
I don't think a temporary is ever needed here, because the type identifier has to be an immutable identifier for the predicate to be considered known.